lib/resty/redis.lua

```diff
@@ -214,7 +214,11 @@ local function _gen_req(args)
     for i = 1, nargs do
         local arg = args[i]
         if type(arg) ~= "string" then
-            arg = tostring(arg)
+            if type(arg) == "number" and arg == math.floor(arg) then 
+                arg = string.format("%.0f",arg)
+            else
+                arg = tostring(arg)
+            end
```

Comment:
I'm afraid this is hacky and awful. We surely do not want to do this in all our Lua library APIs which do a `tostring()` conversion. Note: this redis API really accepts string typed values per se. Better use a string for big numbers in your own Lua code before feeding it into this API function.